src/decompose.cpp

```diff
@@ -27,7 +27,7 @@ alpha_beta get_alpha_beta(const operation& op)
 
 struct find_dot_add
 {
-    auto matcher() const { return match::name("dot", "quant_dot")(match::nargs(3)); }
+    auto matcher() const { return match::name("quant_dot")(match::nargs(3)); }
```

Comment:
I think we may also need to remove the alpha and beta for the quant_dot, but that can be done in another PR.... You can talk to Paul about it.

Reply:
I can do it in another PR. Quant dot in particular requires handling datatype in few places. In most places dot and quant_dot shares the logic.